tree-sitter/src/TreeSitter/Unmarshal.hs

```diff
@@ -221,12 +255,9 @@ instance GUnmarshal f => GUnmarshal (M1 C c f) where
 instance GUnmarshal U1 where
   gunmarshalNode _ = pure U1
 
--- For regular leaf nodes
-instance {-# OVERLAPPABLE #-} GUnmarshal (M1 S s (K1 c Text.Text)) where
-  gunmarshalNode node = M1 . K1 <$> unmarshalNodes [node]
```

Comment:
Why don’t we need this anymore?

Reply:
Regular leaf nodes have multiple fields now, so they look just like products. We therefore handle the distinction via improvements to the `Unmarshal` instance for `Text` instead (and similarly for the annotation types).

Reply:
Aha 👍